packages/mobile/src/services/offline-downloader/offline-downloader.ts

```diff
@@ -102,74 +100,102 @@ export const downloadAllFavorites = async () => {
-  })
+  batchDownloadCollection(favoritedCollections, true)
 }
 
-export const downloadCollection = async (
-  collection: CollectionMetadata,
-  isFavoritesDownload?: boolean,
-  skipTracks?: boolean
+export const batchDownloadCollection = (
+  collections: CollectionMetadata[],
+  isFavoritesDownload: boolean,
+  skipTracks = false
 ) => {
-  const state = store.getState()
-  const currentUserId = getUserId(state)
-
-  // Prevent download of unavailable collections
-  if (
-    !collection ||
-    // @ts-ignore shouldn't be necessary, but not sure we trust the types all the way down
-    collection.variant === Variant.SMART ||
-    collection.is_delete ||
-    (collection.is_private && collection.playlist_owner_id !== currentUserId)
+  const collectionsForDownload: CollectionForDownload[] = collections.map(
+    (collection) => ({
+      collectionId: collection.playlist_id,
+      isFavoritesDownload
+    })
   )
-    return
-
-  const user = getUserFromCollection(state, { id: collection?.playlist_id })
-  const collectionIdStr: string = collection.playlist_id.toString()
   store.dispatch(
-    addCollection({
-      collectionId: collectionIdStr,
-      isFavoritesDownload: !!isFavoritesDownload
+    batchStartCollectionDownload({
+      collectionIds: collectionsForDownload.map(
+        ({ collectionId }) => collectionId
+      ),
+      isFavoritesDownload
     })
   )
+  collectionsForDownload.forEach((collectionForDownload) =>
+    enqueueCollectionDownload(collectionForDownload)
+  )
 
-  if (!user) return
+  if (skipTracks) return
+  const tracksForDownload = collections.flatMap((collection) =>
+    collection.playlist_contents.track_ids.map(({ track: trackId }) => ({
+      trackId,
+      downloadReason: {
+        is_from_favorites: isFavoritesDownload,
+        collection_id: collection.playlist_id.toString()
+      }
+    }))
+  )
+  batchDownloadTrack(tracksForDownload)
+}
+
+export const downloadCollection = async ({
+  collectionId,
+  isFavoritesDownload
+}: CollectionForDownload) => {
+  const state = store.getState()
+  const currentUserId = getUserId(state)
+  store.dispatch(startCollectionDownload({ collectionId, isFavoritesDownload }))
 
-  const isOwner = currentUserId === user.user_id
-  if (!collection.has_current_user_saved && !isOwner) {
+  // Throw this
+  const failJob = (message?: string) => {
     store.dispatch(
-      saveCollection(collection.playlist_id, FavoriteSource.OFFLINE_DOWNLOAD)
+      errorCollectionDownload({ collectionId, isFavoritesDownload })
     )
+    return new Error(message)
   }
 
-  const collectionWithUser: UserCollectionMetadata = {
-    ...collection,
-    user
+  const collectionFromApi = (
+    await apiClient.getPlaylist({
+      playlistId: collectionId,
+      currentUserId
+    })
+  )[0]
+
+  if (!collectionFromApi) {
+    throw failJob(
+      `collection to download not found on discovery - ${collectionFromApi}`
```

Comment:
heads up, i am working on a PR to throw custom typed error messages here, so we can do different things on different kinds of errors